lib/y2logsstep.pm

```diff
@@ -57,61 +58,130 @@ sub get_to_console() {
     }
 }
 
-# to workaround dep issues
-sub record_dependency_issues {
-    while (check_screen 'dependancy-issue', 5) {
-        wait_screen_change {
-            if (check_var('VIDEOMODE', 'text')) {
-                send_key 'alt-s';
-            }
-            else {
-                send_key 'alt-1';
-            }
-        };
-        wait_screen_change {
+# to workaround dependency issues
+sub workaround_dependency_issues {
+    if (check_var('VIDEOMODE', 'text')) {
+        while (check_screen('dependency-issue-text', 5)) {
+            send_key 'alt-s';
+            send_key 'ret';
+            send_key 'alt-o';
+            wait_still_screen 1;
+        }
+    }
+    else {
+        while (check_screen('dependency-issue', 5)) {
+            send_key 'alt-1';
             send_key 'spc';
-        };
-        send_key 'alt-o';
+            send_key 'alt-o';
+            wait_still_screen 1;
+        }
     }
+    wait_screen_change { send_key 'alt-a' };
 }
 
-# check for dependency issues, if found, drill down to software selection, take a screenshot, then die
-sub check_and_record_dependency_problems {
+# to break dependency issues
+sub break_dependency {
+    assert_screen 'dependency-issue';
+
+    if (check_var('VIDEOMODE', 'text')) {
+        while (check_screen('dependency-issue-text', 5)) {    # repeat it untill all dependency issues are resolved
+            send_key 'alt-s';                                 # Solution
+            while (check_screen('break-not-highlighted', 5)) {
+                send_key 'down';                              # down to option break dependency
+            }
+            send_key 'ret';                                   # select option break dependency
+            send_key 'alt-o';                                 # OK - Try Again
+            wait_still_screen 1;
+        }
+    }
+    else {
+        while (check_screen('dependency-issue', 5)) {
+            send_key 'alt-2';                                 # 2 is the option to break dependency
+            send_key 'spc';                                   # select it
+            send_key 'alt-o';                                 # OK
+            wait_still_screen 1;
+        }
+    }
```

Comment:
wait_screen_change { send_key 'spc' }; also causes an endless loop, because the screen doesn't change, it is to focus the mouse on the option "OK - Try Again".